docs/managed_vms/endpoints/client/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"time"
+
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/jws"
+)
+
+var (
+	host   = flag.String("host", "", "The API host. Required.")
+	apiKey = flag.String("api-key", "", "Your API key. Required.")
+
+	serviceAccount = flag.String("service-account", "", "Path to service account JSON file. Cannot be used with -oauth-config.")
+	oauthConfig    = flag.String("oauth-config", "", "Path to oauth client secrets JSON file. Cannot be used with -service-account.")
+)
+
+func main() {
+	flag.Parse()
+
+	if *apiKey == "" || *host == "" {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+	if *serviceAccount == "" && *oauthConfig == "" {
+		fmt.Fprintln(os.Stderr, "Provide one of -oauth-config and -service-account")
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+	if *serviceAccount != "" && *oauthConfig != "" {
+		fmt.Fprintln(os.Stderr, "Can't use -oauth-config and -service-account")
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	var resp *http.Response
+	var err error
+	if *serviceAccount != "" {
+		resp, err = doJWT()
+	} else {
+		log.Fatal("Not implemented.")
```

Comment:
Removed the id_token flow (need to think about how to do this from Go. We don't expose the id_token field in the oauth library).